### PR TITLE
Fix incorrect displayed username `You`(#457)

### PR DESCRIPTION
**Pull request by qzyse2017**
* add email for further check

### CODE REVIEW

**Comment by qzyse2017** [comment]
tested it manually and it performed correctly. Added a new function named ```getCurrentUserEmail``` similar to a previous function at [getCurrentUsername in src/gitService.ts](https://github.com/eamodio/vscode-gitlens/blob/master/src/gitService.ts#L930)

File: src/gitService.ts [hunk 1]
```diff
@@ -935,6 +937,17 @@ export class GitService extends Disposable {
...
         return user;
     }
+    // TODO: Clear cache when git config changes
+    private _userEmailMapCache: Map<string, string | undefined> = new Map();
+
+    async getCurrentUserEmail(repoPath: string) {
+        let userEmail = this._userEmailMapCache.get(repoPath);
+        if (userEmail === undefined) {
+            userEmail = await Git.config_get('user.name', repoPath);
+            this._userNameMapCache.set(repoPath, userEmail);
```
Comment (eamodio) [inline src/gitService.ts]: This should be
```ts
this._userEmailMapCache.set(repoPath, userEmail);
```

**Comment by eamodio** [comment]
@qzyse2017 why the revert?

**Comment by qzyse2017** [comment]
I just want to make this commit be a new branch on my fork of this repo, had not notice this revert will also show in the upstream repo, sorry😥

**Comment by qzyse2017** [comment]
I'm going to issue the fix again

**Comment by qzyse2017** [comment]
I issue a new pull request and since it's not on the same branch with this one, it's not showed here.
see [issue 460](https://github.com/eamodio/vscode-gitlens/pull/460)

File: src/gitService.ts (hunk 1 above)
Comment (qzyse2017) [reply src/gitService.ts]: thanks a lot!

**Pull request closed without merging**